### DISCUSSION TITLE
CLI arguments in dap.continue()

### DISCUSSION THREAD

**Discussion by lars-vc** [mfussenegger/nvim-dap#1047] [Q&A] [answered]
I want to debug a program that takes cli arguments. I see that this is possible using run, but run doesn't search for the current file's configuration, and continue doesn't take any options. So is there a clean way of doing this?
My idea (take this as semi psuedo code) was something like this:
```lua
vim.keymap.set("n", "<leader>dq", function()
    local args = vim.fn.input("Cli arguments: ")
    require("dap").continue({
        args = args,
    })
end, { noremap = true })
```  

https://github.com/mfussenegger/nvim-dap/discussions/597#discussioncomment-3098086
I could find a way to prompt the cli args in the configuration itself, but then it would need to be added to each configuration etc. 
So I am mainly wondering if there is a cleaner way.

**Comment by mfussenegger** [comment, marked as answer]
>  but then it would need to be added to each configuration

If you really want to add the same args to all configurations, it's just a bit of lua:

```lua
local dap = require("dap")
for ft, configs in pairs(dap.configurations) do
    for _, config in ipairs(configs) do
        config.args = args
    end
end
```

The usual approach is to have one dedicated configuration that prompts for args. For example:

```lua
    {
      type = 'python';
      request = 'launch';
      name = 'Launch file with arguments';
      program = '${file}';
      args = function()
        local args_string = vim.fn.input('Arguments: ')
        return vim.split(args_string, " +")
      end;
    }
```

**Reply by lars-vc** [reply]
Oh I didn't realize you can make multiple configs for 1 language and then pick the config. Thanks a lot for the swift answer!

**Comment by lars-vc** [comment]
Mmmh, one thing though, maybe this is a bug: When running a config, terminating and then running run_last, it will run that previous config without reprompting arguments. However, if you then run terminate and run continue then terminate and then run_last it will reprompt every time.
Step by step:
- continue
- terminate
- run_last, all is good, no reprompt
- continue
- terminate
- run_last, reprompts!
